FIX Fixes bug OneHotEncoder's drop_idx_ when there are infrequent categories (#25589)
scikit-learn/scikit-learn: doc/whats_new/v1.2.rst

:mod:`sklearn.preprocessing`
............................

> - |Fix| Fixes a bug in :class:`preprocessing.OneHotEncoder` such that
>   `drop_idx_` properly references the dropped category in the `categories_`
>   attribute when there are infrequent categories. :pr:`25589` by `Thomas Fan`_.

Reviewer: Nit: how about this active form phrasing?
```suggestion
- |Fix| :meth:`preprocessing.OneHotEncoder.drop_idx_` now properly 
  references the dropped category in the `categories_` attribute
  when there are infrequent categories. :pr:`25589` by `Thomas Fan`_.
```

Reviewer 2: Instead of `:meth:`, it should be `:attr:` I THINK.

Reviewer: Thanks! I did not know that this directive existed.
